src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/entity/induction/PreviousWorkExperiencesEntity.kt

```diff
@@ -51,7 +51,7 @@ class PreviousWorkExperiencesEntity(
   @field:NotNull
   var reference: UUID? = null,
 
-  var hasWorkedBefore: Boolean? = null,
+  var hasWorkedBefore: HasWorkedBeforeEntity? = null,
```

Comment:
I think you're going to need a couple of annotations here:
```
@field:NotNull
@Enumerated(value = EnumType.STRING)
```
I think the `@field:NotNull` is largely syntactic sugar. It's a validation annotation, but we don't (currently) validate the entities, instead relying on JSR303 validation on the REST API request.
Having said that I think it's useful to have it on the class if nothing else as a hint to other devs that this field is not intended to be null (and we probably have a non-null constraint on the table in it's original DDL 🤔 )

The other annotation, `@Enumerated(value = EnumType.STRING)`, is more important.
Without it we'll be storing the enum's ordinal index in the database column, rather than it's name. We want the enum name stored in the database, because it then makes more sense when querying the database directly (also remembering that the entire database is exported to the Reporting Platform. They don't need to know that 0 == YES, 1 == NO and 2 == NOT_RELEVANT 🤣 )

Reply:
There is no not-null constraint currently on this field although the mapping I did would set all nulls to NO. I prefer it isn't null in the kotlin type also if it's not null in the DB. Don't know why others don't do that?

Reply:
`@field:NotNull` but also `HasWorkedBeforeEntity? = null,` 🤔 😂

Reply:
The reason we don't make the fields non-nullable (IIRC 🤔 ) is because of the way JPA works, specifically when retrieving records from the database and creating entity class instances. IIRC it uses reflection and (I think) uses a no-args constructor (even though we dont have one - we use the `kotlin-jpa` plugin which provides a no-args constructor) and reflection to set the individual fields. I think I remember this correctly (from a previous non-MoJ project a while back), but I think there are some circumstances where if the fields are non-nullable kotlin gets very grumpy! 🤣 

You'll also notice our entity classes are regular kotlin classes, and not `data` classes. Again, this was specific best-practice all relating to using kotlin classes with JPA.

See an [example here](https://github.com/JetBrains/jpa-buddy-kotlin-entities) (even though they give an example of `data` classes - I'm sure we had trouble with that), but notice the use of `nullable = false` on the `@Column` annotation, even though the kotlin field is nullable (and ` var` rather than `val` so it's value can be set after construction)

Reply:
I forgot to mention above, the best-practice I'm referring to was from [JPA Buddy](https://jpa-buddy.com/) which is a fantastic plug-in for intellij and will generate your entity classes etc (and their website also has a load of guides and best practice recommendations on the subject 👍 )